src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -1327,6 +1387,10 @@ public function get_tag() {
 			return null;
 		}
 
+		if ( isset( $this->current_element ) ) {
+			return $this->current_element->token->node_name;
+		}
```

Comment:
This is what breaks the `tag_name` for PI lookalikes. The node_name is `#comment`, but it's a special comment type that has more handling in the Tag Processor:

https://github.com/WordPress/wordpress-develop/blob/fbb5020a92735e1af1c1873ba6be000a8c19f0dd/src/wp-includes/html-api/class-wp-html-tag-processor.php#L2640-L2651

A potential fix is to only return here if we have a "non-special" node:

```suggestion
		if ( isset( $this->current_element ) && '#' !== $this->current_element->token->node_name[0] ) {
			return $this->current_element->token->node_name;
		}
```